### ISSUE TITLE
Spurious compilation error: not a component of this OBJECT

### ISSUE THREAD

**Issue by nncarlson** [flang-compiler/flang#241]
Consider this example code:
```fortran
module example

  type :: foo
  contains
    procedure :: next => foo_next
  end type

  type :: bar
    type(bar), pointer :: next => null()
  end type

contains

  subroutine foo_next(this)
    class(foo) :: this
  end subroutine

  subroutine fubar
    type(foo) :: obj
    call obj%next
  end subroutine

end module
```
The current version of flang complains incorrectly that `next` is not a component of the local variable `obj` when it is clearly a type bound procedure:
```
$ flang -c flang-bug-20170924a.f90 
F90-S-0142-next is not a component of this OBJECT (flang-bug-20170924a.f90: 20)
  0 inform,   0 warnings,   1 severes, 0 fatal for fubar
```
The spurious error goes away if either 1) the type `bar` is not defined; or 2) if the call includes the unnecessary parens; i.e., 
```fortran
    call obj%next()
```
This issue is related to the ECP ExaAM project.

**Comment by nncarlson** [comment]
This looks to be fixed in the current flang version.
